l1-contracts/contracts/bridgehub/Bridgehub.sol

```diff
@@ -767,6 +767,58 @@ contract Bridgehub is IBridgehub, ReentrancyGuard, Ownable2StepUpgradeable, Paus
         });
     }
 
+    /// @dev Registers an already deployed chain with the bridgehub
+    /// @param _chainId The chain Id of the chain
+    /// @param _hyperchain Address of the hyperchain
+    function registerAlreadyDeployedHyperchain(uint256 _chainId, address _hyperchain) external onlyOwnerOrAdmin onlyL1 {
+        require(_hyperchain != address(0), "BH: hyperchain zero");
+        require(!hyperchainMap.contains(_chainId), "BH: Chain Id Already Exists");
+        require(IZkSyncHyperchain(_hyperchain).getChainId() == _chainId, "BH: Chain Id Mismatch");
+
+        address stm = IZkSyncHyperchain(_hyperchain).getStateTransitionManager();
+        address chainAdmin = IZkSyncHyperchain(_hyperchain).getAdmin();
+        bytes32 chainBaseTokenAssetId = IZkSyncHyperchain(_hyperchain).getBaseTokenAssetId();
+
+        if (_chainId == 0) {
```

Comment:
I think this chainId validation is at least partially repeated during the usual `createNewChain` method, so this logic should be moved out to a separate function